src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -188,7 +173,7 @@ def register_submission(
             "intermediate.documents.report.document",
         )
         execute_unless_result_exists(
-            partial(relate_document, zaak["url"], document["url"]),
+            partial(relate_document, result["zaak"]["url"], document["url"]),
```

Comment:
`zaak` is defined on line 162 already as `zaak = result["zaak"]`

```suggestion
            partial(relate_document, zaak["url"], document["url"]),
```